ibis/backends/duckdb/__init__.py

```diff
@@ -261,30 +172,176 @@ def register(
+
+        self.con.execute(sql)
+        return self._read(table_name)
+
+    def read_parquet(self, source_list, table_name=None, **kwargs):
+        """Register a parquet file as a table in the current database.
+
+        Parameters
+        ----------
+        source_list
+            The data source(s). May be a path to a file or directory of parquet files.
+        table_name
+            An optional name to use for the created table. This defaults to
+            a sequentially generated name.
+        **kwargs
+            Additional keyword arguments passed to DuckDB loading function.
+            See https://duckdb.org/docs/data/parquet for more information.
+
+        Returns
+        -------
+        ir.Table
+            The just-registered table
+        """
+        source_list = normalize_filenames(source_list)
+
+        if any(source.startswith("s3://") for source in source_list):
+            if len(source_list) > 1:
+                raise ValueError("only single s3 paths are supported")
+
+            import pyarrow.dataset as ds
+
+            dataset = ds.dataset(source_list[0])
+            table_name = table_name or f"ibis_read_parquet_{next(paN)}"
+            self._load_extensions(["httpfs"])
             # We don't create a view since DuckDB special cases Arrow Datasets
             # so if we also create a view we end up with both a "lazy table"
             # and a view with the same name
-            with self._safe_raw_sql("SELECT 1") as cursor:
+            with self.con.begin() as con:
                 # DuckDB normally auto-detects Arrow Datasets that are defined
                 # in local variables but the `dataset` variable won't be local
                 # by the time we execute against this so we register it
                 # explicitly.
-                cursor.cursor.c.register(table_name, dataset)
-        elif isinstance(source, (str, Path)):
-            sql, table_name, extensions_required = _generate_view_code(
-                str(source), table_name=table_name, **kwargs
-            )
-            self._load_extensions(extensions_required)
-            self.con.execute(sql)
+                con.connection.c.register(table_name, dataset)
         else:
-            if table_name is None:
-                table_name = next(_gen_table_names)
-            self.con.execute("register", (table_name, source))
+            if any(
+                source.startswith(("http://", "https://")) for source in source_list
+            ):
+                self._load_extensions(["httpfs"])
+            dataset = str(source_list)
+            table_name = table_name or f"ibis_read_parquet_{next(paN)}"
+
+            quoted_table_name = _quote(table_name)
+            sql = f"""CREATE OR REPLACE VIEW {quoted_table_name} AS
+            SELECT * FROM read_parquet({dataset})"""
+
+            self.con.execute(sql)
+
+        return self._read(table_name)
+
+    def read_pandas_or_arrow(self, dataframe, table_name=None):
```

Comment:
Can we either:

1. split this into `read_pandas` and `read_arrow`
2. call this `read_in_memory`/`read_memory`/`read_mem`